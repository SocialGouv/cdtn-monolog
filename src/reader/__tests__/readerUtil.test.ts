import { some } from "fp-ts/Option";

import {
  formatDate,
  getLastDays,
  getLastThreeMonthsComplete,
} from "../readerUtil";

const testDate = new Date(1606227183865);

test("format dates", () => {
  expect(formatDate(testDate)).toBe("2020-11-24");
});

test("get last days", () => {
  expect(getLastDays(2, testDate)).toStrictEqual(["2020-11-23", "2020-11-22"]);
});

test("get last months", () => {
  expect(getLastThreeMonthsComplete(some(testDate))).toMatchSnapshot();
});
