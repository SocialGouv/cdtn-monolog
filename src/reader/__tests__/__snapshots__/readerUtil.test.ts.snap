// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get last months 1`] = `
Array [
  Array [
    "2020-10-01",
    "2020-10-02",
    "2020-10-03",
    "2020-10-04",
    "2020-10-05",
    "2020-10-06",
    "2020-10-07",
    "2020-10-08",
    "2020-10-09",
    "2020-10-10",
    "2020-10-11",
    "2020-10-12",
    "2020-10-13",
    "2020-10-14",
    "2020-10-15",
    "2020-10-16",
    "2020-10-17",
    "2020-10-18",
    "2020-10-19",
    "2020-10-20",
    "2020-10-21",
    "2020-10-22",
    "2020-10-23",
    "2020-10-24",
    "2020-10-25",
    "2020-10-26",
    "2020-10-27",
    "2020-10-28",
    "2020-10-29",
    "2020-10-30",
    "2020-10-31",
  ],
  Array [
    "2020-09-01",
    "2020-09-02",
    "2020-09-03",
    "2020-09-04",
    "2020-09-05",
    "2020-09-06",
    "2020-09-07",
    "2020-09-08",
    "2020-09-09",
    "2020-09-10",
    "2020-09-11",
    "2020-09-12",
    "2020-09-13",
    "2020-09-14",
    "2020-09-15",
    "2020-09-16",
    "2020-09-17",
    "2020-09-18",
    "2020-09-19",
    "2020-09-20",
    "2020-09-21",
    "2020-09-22",
    "2020-09-23",
    "2020-09-24",
    "2020-09-25",
    "2020-09-26",
    "2020-09-27",
    "2020-09-28",
    "2020-09-29",
    "2020-09-30",
  ],
  Array [
    "2020-08-01",
    "2020-08-02",
    "2020-08-03",
    "2020-08-04",
    "2020-08-05",
    "2020-08-06",
    "2020-08-07",
    "2020-08-08",
    "2020-08-09",
    "2020-08-10",
    "2020-08-11",
    "2020-08-12",
    "2020-08-13",
    "2020-08-14",
    "2020-08-15",
    "2020-08-16",
    "2020-08-17",
    "2020-08-18",
    "2020-08-19",
    "2020-08-20",
    "2020-08-21",
    "2020-08-22",
    "2020-08-23",
    "2020-08-24",
    "2020-08-25",
    "2020-08-26",
    "2020-08-27",
    "2020-08-28",
    "2020-08-29",
    "2020-08-30",
    "2020-08-31",
  ],
]
`;
